src/EnergyPlus/HeatingCoils.cc

```diff
@@ -2611,16 +2724,29 @@ namespace HeatingCoils {
             SourceID = HeatingCoil(CoilNum).ReclaimHeatingSourceIndexNum;
             //   Refrigerated cases are simulated at the zone time step, do not decrement available capacity
             //   (the heat reclaim available capacity will not get reinitialized as the air loop iterates)
+            int DesuperheaterNum = CoilNum - NumElecCoil - NumElecCoilMultiStage - NumFuelCoil - NumGasCoilMultiStage;
             if (HeatingCoil(CoilNum).ReclaimHeatingSource == COMPRESSORRACK_REFRIGERATEDCASE) {
-                HeatReclaimRefrigeratedRack(SourceID).UsedHVACCoil = HeatingCoilLoad;
+                HeatReclaimRefrigeratedRack(SourceID).HVACDesuperheaterReclaimedHeat(DesuperheaterNum) = HeatingCoilLoad;
+                HeatReclaimRefrigeratedRack(SourceID).HVACDesuperheaterReclaimedHeatTotal = 0.0;
+                for (auto& num : HeatReclaimRefrigeratedRack(SourceID).HVACDesuperheaterReclaimedHeat) 
+                    HeatReclaimRefrigeratedRack(SourceID).HVACDesuperheaterReclaimedHeatTotal += num;
             } else if (HeatingCoil(CoilNum).ReclaimHeatingSource == CONDENSER_REFRIGERATION) {
-                HeatReclaimRefrigCondenser(SourceID).UsedHVACCoil = HeatingCoilLoad;
+                HeatReclaimRefrigCondenser(SourceID).HVACDesuperheaterReclaimedHeat(DesuperheaterNum) = HeatingCoilLoad;
+                HeatReclaimRefrigCondenser(SourceID).HVACDesuperheaterReclaimedHeatTotal = 0.0;
+                for (auto& num : HeatReclaimRefrigCondenser(SourceID).HVACDesuperheaterReclaimedHeat) 
+                    HeatReclaimRefrigCondenser(SourceID).HVACDesuperheaterReclaimedHeatTotal += num;
             } else if (HeatingCoil(CoilNum).ReclaimHeatingSource == COIL_DX_COOLING ||
                        HeatingCoil(CoilNum).ReclaimHeatingSource == COIL_DX_MULTISPEED ||
                        HeatingCoil(CoilNum).ReclaimHeatingSource == COIL_DX_MULTIMODE) {
-                HeatReclaimDXCoil(SourceID).AvailCapacity -= HeatingCoilLoad;
```

Comment:
If bug successfully fixed in these lines, CI results should reflect large diffs in desuperheaters connected with DX coils, the same applying to water heating desuperheaters + DX coils.